src/libraries/System.Private.Xml/src/System/Xml/Schema/XdrBuilder.cs

```diff
@@ -1655,7 +1655,7 @@ private static void ParseDtMaxLength(ref uint cVal, object obj, XdrBuilder build
                 builder.SendValidationEvent(SR.Sch_DupDtMaxLength);
             }
 
-            if (!ParseInteger((string)obj, ref cVal) || cVal < 0)
+            if (!ParseInteger((string)obj, ref cVal))
```

Comment:
Just to confirm, the intent here wasn't to check that `cValue <= int.MaxValue`, right?

Reply:
I'm not sure what the intent was, but it's been this way for a very long time, so I'd be hesitant to add a new condition without other motivation.